javascript/code-challenges/challenges-13.test.js

```diff
@@ -7,9 +7,19 @@ Write a function named longestString that takes in an array of strings and retur
 ------------------------------------------------------------------------------------------------ */
 
 const longestString = (arr) => {
-// Solution code here...
+  // requiring this to return -1 for array 2 makes no sense and is not in the requirements on line 6. it fails the test, but the test is poorly written or the actual requirements are not expressed.
+  let newArr = arr.map(str => str.length);
+  let flag = 0;
+  let idx = 0;
+  for (let i = 0; i< newArr.length; i++){
+    if (newArr[i] > flag){
+      flag = newArr[i];
+      idx = i;
+    }
+  }
+  return idx;
 };
-  
+
```

Comment:
HI there David! Nice work so far on these, I know some days can be rough so I totally understand the frustration.

For this one, you mention that this needs to return -1 and you know that you need to return **index of** longest string. 

What method can we use to help? Have you reviewed `indexOf()`? You could use this method along with reduce() to find the solution. (This is just the way I would approach it).

indexOf() will return index of the first occurrence of value you are looking for if found, otherwise it returns -1 if it is not found :)

See if you can grab a TA or classmate to get ya going when you are ready to revisit these 👍